extensions/amp-mega-menu/0.1/amp-mega-menu.js

```diff
@@ -0,0 +1,322 @@
+    const index = this.items_.indexOf(item);
+    if (index !== -1) {
+      event.preventDefault();
+      let dir = event.key == Keys.LEFT_ARROW ? -1 : 1;
+      // Left is 'previous' in LTR and 'next' in RTL; vice versa for Right.
+      if (isRTL(this.document_)) {
+        dir = -dir;
+      }
+      // If user navigates one past the beginning or end, wrap around.
+      let newIndex = (index + dir) % this.items_.length;
+      if (newIndex < 0) {
+        newIndex += this.items_.length;
+      }
+      tryFocus(this.getItemHeading_(this.items_[newIndex]));
+    }
+  }
+
+  /**
+   * Expand the given item to show its menu content.
+   * @param {!Element} item
+   */
+  expand_(item) {
+    this.mutateElement(() => {
+      // Wait for mutateElement, so that the element has been transfered to the
+      // fixed layer. This is needed to hide the correct elements.
+      const content = this.getItemContent_(item);
+      setModalAsOpen(content);
+      content.setAttribute('aria-modal', 'true');
+    });
+    item.setAttribute('open', '');
+    this.maskElement_.setAttribute('open', '');
+    const heading = this.getItemHeading_(item);
+    heading.setAttribute('aria-expanded', 'true');
+    const screenReaderCloseButton = dev().assertElement(
+      item.querySelector('.i-amphtml-screen-reader')
+    );
+    tryFocus(screenReaderCloseButton);
+    this.expandedItem_ = item;
+  }
+
+  /**
+   * Collapse any expanded item inside the mega menu.
+   * @return {?Element} the item that was previously expanded, if any.
+   */
+  collapse_() {
+    if (!this.expandedItem_) {
+      return null;
+    }
+    const item = dev().assertElement(this.expandedItem_);
+    this.mutateElement(() => {
+      const content = this.getItemContent_(item);
+      setModalAsClosed(content);
+      content.setAttribute('aria-modal', 'false');
+    });
+    item.removeAttribute('open');
+    this.maskElement_.removeAttribute('open');
+    const heading = this.getItemHeading_(item);
+    heading.setAttribute('aria-expanded', 'false');
+    tryFocus(heading);
+    this.expandedItem_ = null;
+    return item;
+  }
+
+  /**
+   * Return the heading of given menu item.
+   * @param {!Element} item
+   * @return {!Element}
+   */
+  getItemHeading_(item) {
+    const heading = scopedQuerySelector(item, '> .i-amphtml-mega-menu-heading');
+    return dev().assertElement(heading);
+  }
+
+  /**
+   * Return the expandable content of given menu item.
+   * @param {!Element} item
+   * @return {!Element}
+   */
+  getItemContent_(item) {
+    const content = scopedQuerySelector(item, '> .i-amphtml-mega-menu-content');
+    return dev().assertElement(content);
+  }
+
+  /**
+   * Creates an "invisible" close button for screen readers to collapse the
+   * mega menu.
+   * @return {!Element}
+   */
+  createScreenReaderCloseButton() {
+    const ariaLabel =
+      this.element.getAttribute('data-close-button-aria-label') ||
+      'Close the mega menu';
```

Comment:
Pardon my lack of familiarity with some `amphtml` conventions, but do we have any approach to internationalization for text strings like this (even though there's a way to set it via attribute)?

Irrespective of that, maybe this would make sense as a constant.

Reply:
afaik there's no unified approach to internationalizing these strings; I can make this a constant.